vendor/k8s.io/kubernetes/plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -80,26 +82,35 @@ func (l *lifecycle) Admit(a admission.Attributes) (err error) {
 		return nil
 	}
 
-	namespaceObj, exists, err := l.store.Get(&api.Namespace{
-		ObjectMeta: api.ObjectMeta{
-			Name:      a.GetNamespace(),
-			Namespace: "",
-		},
-	})
+	namespaceObj, exists, err := l.store.Get(&api.Namespace{ObjectMeta: api.ObjectMeta{Name: a.GetNamespace()}})
 	if err != nil {
 		return errors.NewInternalError(err)
 	}
 
+	if !exists {
+		// give the cache time to observe a recent namespace creation
+		time.Sleep(50 * time.Millisecond)
```

Comment:
this is also not just namespace creation.  if a namespace is deleted, we purge it from the local store.